planner/core/logical_plan_builder.go

```diff
@@ -2536,7 +2536,9 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	if err != nil {
 		return nil, err
 	}
-	possiblePaths, err = b.filterPathByIsolationRead(possiblePaths)
+	if dbName.L != mysql.SystemDB {
```

Comment:
Can we put the check into the function? In case that someone else called the function but forgot the check.